src/Qir/Execution/Tools/Driver/QirCppDriver.tt

```diff
@@ -0,0 +1,246 @@
+﻿<#@ template language="C#" linePragmas="false" #>
+<#@ import namespace="System.IO" #>
+<#@ import namespace="Microsoft.Quantum.QsCompiler.BondSchemas.Execution" #>
+//------------------------------------------------------------------------------
+// This code was generated by a tool.
```

Comment:
Hard to understand if this file was generated by a tool or if this the source templated used by a tool and generated files will have this comment. Think like a person who sees this codebase for the first time and this is the first file that person sees.

Reply:
Created a README.md file at the *src/Qir/Execution/Tools* level that explains the *\*.tt* files and the corresponding generated *\*.cs* files. It also describes a bit more in detail the contents of each subfolder.